packages/util-rxjs/test/blockingWithLatestFrom.test.ts

```diff
@@ -0,0 +1,36 @@
+/* eslint-disable no-multi-spaces */
+
+import { blockingWithLatestFrom } from '../src';
+import { createTestScheduler } from '@cardano-sdk/util-dev';
+import { firstValueFrom, from, of, toArray } from 'rxjs';
+
+/* eslint-disable prettier/prettier */
+describe('blockingWithLatestFrom', () => {
+  it('waits for dependency$ to emit, does not lose any source emissions and subscribes only once', () => {
+    createTestScheduler().run(({ hot, expectObservable, expectSubscriptions }) => {
+      const source$ =     hot('a--b-c');
```

Comment:
What is the expected behaviour when `source$` emits multiple values before `dependency$` emits?
Maybe add a test for it too.

Reply:
It will defer any emissions until `dependency$` emits and then emit all of them combined with the dependency

Reply:
Added some comments and updated the test in 797159e30af9e06843643d7bb593f9f9c9d7d35d